jfix-armeria-aggregating-profiler/src/main/kotlin/ru/fix/armeria/aggregating/profiler/ProfiledHttpClient.kt

```diff
@@ -0,0 +1,172 @@
+                        protocol = log.sessionProtocol(),
+                        channel = log.channel()
+                    )
+                )
+                /*
+                 * USEFUL INFORMATION
+                 * it has been discovered by tests, that
+                 * RequestOnlyLog.requestStartTime always <= RequestOnlyLog.connectionTimings().connectionAcquisitionStartTime
+                 *
+                 * That is why first value is used here to profile time from the absolute request start to a moment
+                 * when connection acquired.
+                 */
+            ).call(log.requestStartTimeMillis())
+        }
+    }
+
+    private fun profileOnRequestCompleted(req: HttpRequest, log: RequestLog) {
+        val profiledError: ProfiledError? = log.detectProfiledErrorIfAny()
+
+        if (profiledError != null && profiledError is ProfiledError.UnrecognizedError) {
+            when (val unrecognizedErrorType = profiledError) {
+                is ProfiledError.UnrecognizedError.UnknownStatus -> {
+                    logger.error {
+                        "Unexpected UNKNOWN response status of request log $log"
+                    }
+                    return
+                }
+                is ProfiledError.UnrecognizedError.WithCause -> {
+                    val causeSource = when (unrecognizedErrorType) {
+                        is ProfiledError.UnrecognizedError.WithCause.InRequest -> "request"
+                        is ProfiledError.UnrecognizedError.WithCause.InResponse -> "response"
+                    }
+                    logger.warn(unrecognizedErrorType.cause) { "Unrecognized $causeSource exception occurred: " }
+                }
+            }
+        }
+
+        val tags = MetricTags.build(
+            path = req.path(),
+            method = req.method(),
+            remoteInetSocketAddress = log.context().endpointInetSockedAddress,
+            responseStatusCode = log.knownResponseStatus,
+            protocol = log.sessionProtocol,
+            channel = log.channel()
+        )
+
+        when {
+            profiledError != null -> {
+
+                profiler.profiledCall(
+                    Identity(Metrics.HTTP_ERROR, tags + profiledError.metricTags)
+                ).let {
+                    if (profiledError.latencyMetricRequired) {
+                        it.call(log.requestStartTimeMillis())
+                    } else {
+                        it.call()
+                    }
+                }
+            }
+            else -> {
+
+                profiler.profiledCall(
+                    Identity(Metrics.HTTP_SUCCESS, tags)
+                ).call(log.requestStartTimeMillis())
+            }
+        }
+
+        log.connectionTimings()?.let {
+            logger.debug {
+                """Request log = $log
+                    |connections timings = $it""".trimMargin()
+            }
```

Comment:
looks complex, can be simplified like this:
```suggestion
            logger.debug { "Request log = $log connections timings = $it" }
```